src/lib/rng/esdm_rng/esdm_rng.cpp

```diff
@@ -0,0 +1,83 @@
+/*
+* ESDM RNG
+* (C) 2024, Markus Theil <theil.markus@gmail.com>
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/esdm_rng.h>
+
+#include <esdm/esdm_rpc_client.h>
+
+namespace Botan {
+
+namespace {
+class ESDM_Context {
+   private:
+      constinit inline static std::mutex g_mutex;
+
+   public:
+      [[nodiscard]] static std::shared_ptr<void> instance() {
+         std::scoped_lock lk(g_mutex);
+         static std::shared_ptr<ESDM_Context> g_instance{new ESDM_Context()};
+         return g_instance->acquire();
+      }
```

Comment:
The `g_mutex` isn't necessary here. As of C++11 it is guaranteed that lazy initialization of local static variables is threadsafe: https://en.cppreference.com/w/cpp/language/storage_duration#Static_block_variables

> If multiple threads attempt to initialize the same static local variable concurrently, the initialization occurs exactly once (similar behavior can be obtained for arbitrary functions with [std::call_once](https://en.cppreference.com/w/cpp/thread/call_once)).

Also: its just fine to not use a smart pointer here but just directly initialize the `ESDM_Context`.

```suggestion
      [[nodiscard]] static std::shared_ptr<void> instance() {
         static ESDM_Context g_instance{};
         return g_instance->acquire();
      }
```